.github/workflows/gradle.yml

```diff
@@ -0,0 +1,99 @@
+name: CI
+on: [push, pull_request]
+
+jobs:
+  build_on_linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        java: [ '8', '11', '14', '15', '15-site', '15-leak']
+    steps:
+    - uses: actions/checkout@v2
+
+    - if: "!startsWith(matrix.java, '15')"
+      name: Set up JRE ${{ matrix.java }}
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: ${{ matrix.java }}
+        java-package: jre
+
+    - if: "!startsWith(matrix.java, '15')"
+      name: Set up JAVA_TEST_HOME env
+      run: |
+        echo "JAVA_TEST_HOME=${JAVA_HOME}" >> $GITHUB_ENV
+        echo "TEST_JAVA_VERSION=${{ matrix.java }}" >> $GITHUB_ENV
+
+    - name: Set up JDK 15
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: '15'
+
+    - name: Cache
```

Comment:
Strongly recommend this action instead much faster / stabler

https://github.com/burrunan/gradle-cache-action